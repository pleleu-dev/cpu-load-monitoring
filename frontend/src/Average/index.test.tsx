import renderer from 'react-test-renderer'

import { Average } from '.'
import type { Load } from '../types'

const data: Load[] = [
  {
    time: 1712299160001,
    average: 0.3,
    formattedTime: '08:39:20',
  },
  {
    time: 1712299156011,
    average: 1.4,
    formattedTime: '08:39:16',
  },
  {
    time: 1712299157021,
    average: 0.5,
    formattedTime: '08:39:17',
  },
  {
    time: 1712299158031,
    average: 2.6,
    formattedTime: '08:39:18',
  },
]

describe('Average', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Average data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
