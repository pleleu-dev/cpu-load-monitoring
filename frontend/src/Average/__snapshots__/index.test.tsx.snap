// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Average should render correctly 1`] = `
<div
  className="chart"
>
  <div>
    <div
      className="recharts-wrapper"
      style={
        {
          "cursor": "default",
          "height": 500,
          "position": "relative",
          "width": 500,
        }
      }
    >
      <svg
        className="recharts-surface"
        height={500}
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
        viewBox="0 0 500 500"
        width={500}
      >
        <title />
        <desc />
        <defs>
          <clipPath
            id="recharts1-clip"
          >
            <rect
              height={460}
              width={430}
              x={65}
              y={5}
            />
          </clipPath>
        </defs>
        <g
          className="recharts-layer recharts-line"
        >
          <path
            className="recharts-curve recharts-line-curve"
            clipPath={null}
            d="M65,411.9230769230769C112.77777777777777,314.6153846153846,160.55555555555557,217.3076923076923,208.33333333333334,217.3076923076923C256.11111111111114,217.3076923076923,303.8888888888889,376.53846153846155,351.6666666666667,376.53846153846155C399.44444444444446,376.53846153846155,447.22222222222223,190.7692307692308,495,5"
            fill="none"
            height={460}
            stroke=""
            strokeDasharray="0px 0px"
            strokeWidth={1}
            width={430}
          />
        </g>
        <g
          className="recharts-cartesian-grid"
        >
          <g
            className="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height={460}
              stroke=""
              strokeDasharray="5 5"
              width={430}
              x={65}
              x1={65}
              x2={495}
              y={5}
              y1={465}
              y2={465}
            />
            <line
              fill="none"
              height={460}
              stroke=""
              strokeDasharray="5 5"
              width={430}
              x={65}
              x1={65}
              x2={495}
              y={5}
              y1={350}
              y2={350}
            />
            <line
              fill="none"
              height={460}
              stroke=""
              strokeDasharray="5 5"
              width={430}
              x={65}
              x1={65}
              x2={495}
              y={5}
              y1={235}
              y2={235}
            />
            <line
              fill="none"
              height={460}
              stroke=""
              strokeDasharray="5 5"
              width={430}
              x={65}
              x1={65}
              x2={495}
              y={5}
              y1={120}
              y2={120}
            />
            <line
              fill="none"
              height={460}
              stroke=""
              strokeDasharray="5 5"
              width={430}
              x={65}
              x1={65}
              x2={495}
              y={5}
              y1={5}
              y2={5}
            />
          </g>
          <g
            className="recharts-cartesian-grid-vertical"
          >
            <line
              fill="none"
              height={460}
              stroke=""
              strokeDasharray="5 5"
              width={430}
              x={65}
              x1={65}
              x2={65}
              y={5}
              y1={5}
              y2={465}
            />
            <line
              fill="none"
              height={460}
              stroke=""
              strokeDasharray="5 5"
              width={430}
              x={65}
              x1={208.33333333333334}
              x2={208.33333333333334}
              y={5}
              y1={5}
              y2={465}
            />
            <line
              fill="none"
              height={460}
              stroke=""
              strokeDasharray="5 5"
              width={430}
              x={65}
              x1={351.6666666666667}
              x2={351.6666666666667}
              y={5}
              y1={5}
              y2={465}
            />
            <line
              fill="none"
              height={460}
              stroke=""
              strokeDasharray="5 5"
              width={430}
              x={65}
              x1={495}
              x2={495}
              y={5}
              y1={5}
              y2={465}
            />
          </g>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
        >
          <line
            className="recharts-cartesian-axis-line"
            fill="none"
            height={30}
            orientation="bottom"
            stroke=""
            width={430}
            x={65}
            x1={65}
            x2={495}
            y={465}
            y1={465}
            y2={465}
          />
          <g
            className="recharts-cartesian-axis-ticks"
          >
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                orientation="bottom"
                stroke=""
                width={430}
                x={65}
                x1={65}
                x2={65}
                y={465}
                y1={471}
                y2={465}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill=""
                height={30}
                orientation="bottom"
                stroke="none"
                textAnchor="middle"
                width={430}
                x={65}
                y={473}
              >
                <tspan
                  dy="0.71em"
                  x={65}
                >
                  08:39:20
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                orientation="bottom"
                stroke=""
                width={430}
                x={65}
                x1={208.33333333333334}
                x2={208.33333333333334}
                y={465}
                y1={471}
                y2={465}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill=""
                height={30}
                orientation="bottom"
                stroke="none"
                textAnchor="middle"
                width={430}
                x={208.33333333333334}
                y={473}
              >
                <tspan
                  dy="0.71em"
                  x={208.33333333333334}
                >
                  08:39:16
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                orientation="bottom"
                stroke=""
                width={430}
                x={65}
                x1={351.6666666666667}
                x2={351.6666666666667}
                y={465}
                y1={471}
                y2={465}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill=""
                height={30}
                orientation="bottom"
                stroke="none"
                textAnchor="middle"
                width={430}
                x={351.6666666666667}
                y={473}
              >
                <tspan
                  dy="0.71em"
                  x={351.6666666666667}
                >
                  08:39:17
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                orientation="bottom"
                stroke=""
                width={430}
                x={65}
                x1={495}
                x2={495}
                y={465}
                y1={471}
                y2={465}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill=""
                height={30}
                orientation="bottom"
                stroke="none"
                textAnchor="middle"
                width={430}
                x={495}
                y={473}
              >
                <tspan
                  dy="0.71em"
                  x={495}
                >
                  08:39:18
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
        >
          <line
            className="recharts-cartesian-axis-line"
            fill="none"
            height={460}
            orientation="left"
            stroke=""
            width={60}
            x={5}
            x1={65}
            x2={65}
            y={5}
            y1={5}
            y2={465}
          />
          <g
            className="recharts-cartesian-axis-ticks"
          >
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={460}
                orientation="left"
                stroke=""
                width={60}
                x={5}
                x1={59}
                x2={65}
                y={5}
                y1={465}
                y2={465}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill=""
                height={460}
                orientation="left"
                stroke="none"
                textAnchor="end"
                width={60}
                x={57}
                y={465}
              >
                <tspan
                  dy="0.355em"
                  x={57}
                >
                  0
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={460}
                orientation="left"
                stroke=""
                width={60}
                x={5}
                x1={59}
                x2={65}
                y={5}
                y1={350}
                y2={350}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill=""
                height={460}
                orientation="left"
                stroke="none"
                textAnchor="end"
                width={60}
                x={57}
                y={350}
              >
                <tspan
                  dy="0.355em"
                  x={57}
                >
                  0.65
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={460}
                orientation="left"
                stroke=""
                width={60}
                x={5}
                x1={59}
                x2={65}
                y={5}
                y1={235}
                y2={235}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill=""
                height={460}
                orientation="left"
                stroke="none"
                textAnchor="end"
                width={60}
                x={57}
                y={235}
              >
                <tspan
                  dy="0.355em"
                  x={57}
                >
                  1.3
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={460}
                orientation="left"
                stroke=""
                width={60}
                x={5}
                x1={59}
                x2={65}
                y={5}
                y1={120}
                y2={120}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill=""
                height={460}
                orientation="left"
                stroke="none"
                textAnchor="end"
                width={60}
                x={57}
                y={120}
              >
                <tspan
                  dy="0.355em"
                  x={57}
                >
                  1.95
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={460}
                orientation="left"
                stroke=""
                width={60}
                x={5}
                x1={59}
                x2={65}
                y={5}
                y1={5}
                y2={5}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill=""
                height={460}
                orientation="left"
                stroke="none"
                textAnchor="end"
                width={60}
                x={57}
                y={5}
              >
                <tspan
                  dy="0.355em"
                  x={57}
                >
                  2.6
                </tspan>
              </text>
            </g>
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
